metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -241,6 +243,27 @@ final class BloopServers(
     }.toOption
   }
 
+  private def maybeLoadBloopGlobalJsonFile(
+      bloopGlobalJsonFilePath: AbsolutePath
+  ): (Option[String], Option[List[String]]) = {
+    val maybeData = Try {
+      val source = scala.io.Source.fromFile(bloopGlobalJsonFilePath.toURI)
+      val jsonString =
+        try source.mkString
+        finally source.close()
+      ujson.read(jsonString)
+    }
+    val javaHome = maybeData.flatMap { data =>
+      Try(data("javaHome").str)
+    }.toOption
+    val javaOptions = maybeData.flatMap { data =>
+      Try {
+        data("javaOptions").arr.toList.map(_.str)
+      }
+    }
```

Comment:
```suggestion
     val javaHome = for {
        obj <- maybeData.objOpt
        strValue <- obj.get("javaHome")
        str <- strValue.strOpt
      } yield str
    val javaOptions = for {
      obj <- maybeData.objOpt
      value <- obj.get("javaOptions")
      arr <- value.arrOpt
      str = arr.flatMap(_.strOpt)
   } yield str
```

No need for the Try this way and it's better to handle things explicitely.

Reply:
applied. Thanks!